File: backend/src/main/java/com/northwollo/tourism/dto/response/NearbyTourismDto.java
package com.northwollo.tourism.dto.response;

import com.northwollo.tourism.entity.TourismPlace;
import lombok.Data;

@Data
public class NearbyTourismDto {
    private Long id;
    private String name;
    private String imageUrl;

    public static NearbyTourismDto fromEntity(TourismPlace place) {
        NearbyTourismDto dto = new NearbyTourismDto();
        dto.setId(place.getId());
        dto.setName(place.getName());
        // ✅ Use first image if available
        dto.setImageUrl(
                place.getImages() != null && !place.getImages().isEmpty() ?
                        place.getImages().get(0).getImageUrl() : null
        );
        return dto;
    }
}

File: backend-nestjs/src/modules/tourism/dto/nearby-tourism.dto.ts
export class NearbyTourismDto {
  id: number;
  name: string;
  imageUrl?: string;

  constructor(id: number, name: string, imageUrl?: string) {
    this.id = id;
    this.name = name;
    this.imageUrl = imageUrl;
  }

  static fromEntity(place: any): NearbyTourismDto {
    return new NearbyTourismDto(
      place.id,
      place.name,
      place.images && place.images.length > 0 ? place.images[0].imageUrl : null,
    );
  }
}
